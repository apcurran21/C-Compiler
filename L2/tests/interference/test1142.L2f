(@main
	0
	:call_label0
	goto :call_label7
	:call_label1
	return
	:call_label2
	rdi <- %arrayAddr
	mem rsp -8 <- :call_label8
	call @array_encode 1
	:call_label8
	rdi <- %arrayAddr
	call print 1
	return
	:call_label3
	%newVar1 <- 0
	%newVar1 += %pos0
	%newVar0 <- %newVar1
	%newVar0 *= 8
	%newVar0 += 24
	%newVar3 <- %arrayAddr
	%newVar3 += %newVar0
	%val0 <- mem %newVar3 0
	%newVar5 <- 0
	%newVar5 += %pos1
	%newVar4 <- %newVar5
	%newVar4 *= 8
	%newVar4 += 24
	%newVar7 <- %arrayAddr
	%newVar7 += %newVar4
	%val1 <- mem %newVar7 0
	%val2 <- %val0
	%val2 += %val1
	%newVar9 <- 0
	%newVar9 += %pos2
	%newVar8 <- %newVar9
	%newVar8 *= 8
	%newVar8 += 24
	%newVar11 <- %arrayAddr
	%newVar11 += %newVar8
	mem %newVar11 0 <- %val2
	%count += 1
	%pos0 += 1
	%pos1 += 1
	%pos2 += 1
	goto :call_label4
	:call_label4
	%ifDone <- %arrayLen < %count
	cjump %ifDone = 1 :call_label2
	goto :call_label3
	:call_label5
	rdi <- %arg1
	mem rsp -8 <- :call_label9
	call @int_encode 1
	:call_label9
	%arg1 <- rax
	%newVar13 <- %arg1
	%newVar13 >>= 1
	%newVar12 <- %newVar13
	%newVar12 <<= 1
	%newVar12 += 1
	%newVar12 += 4
	rdi <- %newVar12
	rsi <- 1
	call allocate 2
	%arrayAddr <- rax
	%newVar14 <- %arrayAddr
	%newVar14 += 8
	mem %newVar14 0 <- 3
	%newVar15 <- %arrayAddr
	%newVar15 += 16
	mem %newVar15 0 <- %arg1
	%pos0 <- 0
	%pos1 <- 1
	%pos2 <- 2
	%newVar17 <- 0
	%newVar17 += %pos0
	%newVar16 <- %newVar17
	%newVar16 *= 8
	%newVar16 += 24
	%newVar19 <- %arrayAddr
	%newVar19 += %newVar16
	mem %newVar19 0 <- 0
	%newVar21 <- 0
	%newVar21 += %pos1
	%newVar20 <- %newVar21
	%newVar20 *= 8
	%newVar20 += 24
	%newVar23 <- %arrayAddr
	%newVar23 += %newVar20
	mem %newVar23 0 <- 1
	%newVar24 <- 0
	%newVar24 *= 8
	%newVar25 <- %newVar24
	%newVar25 += 16
	%newVar26 <- %arrayAddr
	%newVar26 += %newVar25
	%arrayLen <- mem %newVar26 0
	rdi <- %arrayLen
	mem rsp -8 <- :call_label10
	call @int_decode 1
	:call_label10
	%arrayLen <- rax
	%count <- 3
	goto :call_label4
	:call_label6
	%arg1 <- 10
	%invalid <- %arg1 < 2
	cjump %invalid = 1 :call_label1
	goto :call_label5
	:call_label7
	goto :call_label6
)
