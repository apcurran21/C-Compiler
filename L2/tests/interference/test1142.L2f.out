%arg1 %arrayAddr %invalid %newVar12 %newVar13 %newVar14 %newVar15 r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
%arrayAddr %arg1 %arrayLen %count %ifDone %newVar0 %newVar1 %newVar11 %newVar14 %newVar15 %newVar16 %newVar17 %newVar19 %newVar20 %newVar21 %newVar23 %newVar24 %newVar25 %newVar26 %newVar3 %newVar4 %newVar5 %newVar7 %newVar8 %newVar9 %pos0 %pos1 %pos2 r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi %val0 %val1 %val2
%arrayLen %arrayAddr %count %ifDone %newVar0 %newVar1 %newVar11 %newVar3 %newVar4 %newVar5 %newVar7 %newVar8 %newVar9 %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx %val0 %val1 %val2
%count %arrayAddr %arrayLen %ifDone %newVar0 %newVar1 %newVar11 %newVar3 %newVar4 %newVar5 %newVar7 %newVar8 %newVar9 %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx %val0 %val1 %val2
%ifDone %arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx
%invalid %arg1 r12 r13 r14 r15 rax rbp rbx
%newVar0 %arrayAddr %arrayLen %count %newVar3 %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx
%newVar11 %arrayAddr %arrayLen %count %newVar8 %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx %val2
%newVar12 %arg1 r12 r13 r14 r15 rbp rbx
%newVar13 %arg1 r12 r13 r14 r15 rbp rbx
%newVar14 %arg1 %arrayAddr r12 r13 r14 r15 rbp rbx
%newVar15 %arg1 %arrayAddr r12 r13 r14 r15 rbp rbx
%newVar16 %arrayAddr %newVar19 %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx
%newVar17 %arrayAddr %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx
%newVar19 %arrayAddr %newVar16 %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx
%newVar1 %arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx
%newVar20 %arrayAddr %newVar23 %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx
%newVar21 %arrayAddr %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx
%newVar23 %arrayAddr %newVar20 %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx
%newVar24 %arrayAddr %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx
%newVar25 %arrayAddr %newVar26 %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx
%newVar26 %arrayAddr %newVar25 %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx
%newVar3 %arrayAddr %arrayLen %count %newVar0 %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx
%newVar4 %arrayAddr %arrayLen %count %newVar7 %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx %val0
%newVar5 %arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx %val0
%newVar7 %arrayAddr %arrayLen %count %newVar4 %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx %val0
%newVar8 %arrayAddr %arrayLen %count %newVar11 %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx %val2
%newVar9 %arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx %val2
%pos0 %arrayAddr %arrayLen %count %ifDone %newVar0 %newVar1 %newVar11 %newVar16 %newVar17 %newVar19 %newVar20 %newVar21 %newVar23 %newVar24 %newVar25 %newVar26 %newVar3 %newVar4 %newVar5 %newVar7 %newVar8 %newVar9 %pos1 %pos2 r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi %val0 %val1 %val2
%pos1 %arrayAddr %arrayLen %count %ifDone %newVar0 %newVar1 %newVar11 %newVar16 %newVar17 %newVar19 %newVar20 %newVar21 %newVar23 %newVar24 %newVar25 %newVar26 %newVar3 %newVar4 %newVar5 %newVar7 %newVar8 %newVar9 %pos0 %pos2 r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi %val0 %val1 %val2
%pos2 %arrayAddr %arrayLen %count %ifDone %newVar0 %newVar1 %newVar11 %newVar16 %newVar17 %newVar19 %newVar20 %newVar21 %newVar23 %newVar24 %newVar25 %newVar26 %newVar3 %newVar4 %newVar5 %newVar7 %newVar8 %newVar9 %pos0 %pos1 r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi %val0 %val1 %val2
r10 %arg1 %arrayAddr %pos0 %pos1 %pos2 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
r11 %arg1 %arrayAddr %pos0 %pos1 %pos2 r10 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi
r12 %arg1 %arrayAddr %arrayLen %count %ifDone %invalid %newVar0 %newVar1 %newVar11 %newVar12 %newVar13 %newVar14 %newVar15 %newVar16 %newVar17 %newVar19 %newVar20 %newVar21 %newVar23 %newVar24 %newVar25 %newVar26 %newVar3 %newVar4 %newVar5 %newVar7 %newVar8 %newVar9 %pos0 %pos1 %pos2 r10 r11 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi %val0 %val1 %val2
r13 %arg1 %arrayAddr %arrayLen %count %ifDone %invalid %newVar0 %newVar1 %newVar11 %newVar12 %newVar13 %newVar14 %newVar15 %newVar16 %newVar17 %newVar19 %newVar20 %newVar21 %newVar23 %newVar24 %newVar25 %newVar26 %newVar3 %newVar4 %newVar5 %newVar7 %newVar8 %newVar9 %pos0 %pos1 %pos2 r10 r11 r12 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi %val0 %val1 %val2
r14 %arg1 %arrayAddr %arrayLen %count %ifDone %invalid %newVar0 %newVar1 %newVar11 %newVar12 %newVar13 %newVar14 %newVar15 %newVar16 %newVar17 %newVar19 %newVar20 %newVar21 %newVar23 %newVar24 %newVar25 %newVar26 %newVar3 %newVar4 %newVar5 %newVar7 %newVar8 %newVar9 %pos0 %pos1 %pos2 r10 r11 r12 r13 r15 r8 r9 rax rbp rbx rcx rdi rdx rsi %val0 %val1 %val2
r15 %arg1 %arrayAddr %arrayLen %count %ifDone %invalid %newVar0 %newVar1 %newVar11 %newVar12 %newVar13 %newVar14 %newVar15 %newVar16 %newVar17 %newVar19 %newVar20 %newVar21 %newVar23 %newVar24 %newVar25 %newVar26 %newVar3 %newVar4 %newVar5 %newVar7 %newVar8 %newVar9 %pos0 %pos1 %pos2 r10 r11 r12 r13 r14 r8 r9 rax rbp rbx rcx rdi rdx rsi %val0 %val1 %val2
r8 %arg1 %arrayAddr %pos0 %pos1 %pos2 r10 r11 r12 r13 r14 r15 r9 rax rbp rbx rcx rdi rdx rsi
r9 %arg1 %arrayAddr %pos0 %pos1 %pos2 r10 r11 r12 r13 r14 r15 r8 rax rbp rbx rcx rdi rdx rsi
rax %arg1 %arrayAddr %invalid %pos0 %pos1 %pos2 r10 r11 r12 r13 r14 r15 r8 r9 rbp rbx rcx rdi rdx rsi
rbp %arg1 %arrayAddr %arrayLen %count %ifDone %invalid %newVar0 %newVar1 %newVar11 %newVar12 %newVar13 %newVar14 %newVar15 %newVar16 %newVar17 %newVar19 %newVar20 %newVar21 %newVar23 %newVar24 %newVar25 %newVar26 %newVar3 %newVar4 %newVar5 %newVar7 %newVar8 %newVar9 %pos0 %pos1 %pos2 r10 r11 r12 r13 r14 r15 r8 r9 rax rbx rcx rdi rdx rsi %val0 %val1 %val2
rbx %arg1 %arrayAddr %arrayLen %count %ifDone %invalid %newVar0 %newVar1 %newVar11 %newVar12 %newVar13 %newVar14 %newVar15 %newVar16 %newVar17 %newVar19 %newVar20 %newVar21 %newVar23 %newVar24 %newVar25 %newVar26 %newVar3 %newVar4 %newVar5 %newVar7 %newVar8 %newVar9 %pos0 %pos1 %pos2 r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rcx rdi rdx rsi %val0 %val1 %val2
rcx %arg1 %arrayAddr %pos0 %pos1 %pos2 r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rdi rdx rsi
rdi %arg1 %arrayAddr %pos0 %pos1 %pos2 r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdx rsi
rdx %arg1 %arrayAddr %pos0 %pos1 %pos2 r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rsi
rsi %arg1 %arrayAddr %pos0 %pos1 %pos2 r10 r11 r12 r13 r14 r15 r8 r9 rax rbp rbx rcx rdi rdx
%val0 %arrayAddr %arrayLen %count %newVar4 %newVar5 %newVar7 %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx %val1
%val1 %arrayAddr %arrayLen %count %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx %val0 %val2
%val2 %arrayAddr %arrayLen %count %newVar11 %newVar8 %newVar9 %pos0 %pos1 %pos2 r12 r13 r14 r15 rbp rbx %val1
